### PR TITLE
Traits review

### PR DESCRIPTION
**Pull request by sinkingsugar**
Review traits and improve storage

### CODE REVIEW

File: pallets/protos/src/lib.rs [hunk 1]
```diff
@@ -845,25 +867,6 @@ pub mod pallet {
 			let found: Vec<_> = categories
 				.into_iter()
 				.filter(|cat| match cat {
-					Categories::Trait(proto_shard_trait) => {
```
Comment (alebaffa) [inline pallets/protos/src/lib.rs:848]: Though, this removes completely the filter by trait name or id which is a feature that I understood was needed, no?

Comment (alebaffa) [reply pallets/protos/src/lib.rs:848]: Also mentioned here https://github.com/fragcolor-xyz/asset-store/issues/37

Comment (alebaffa) [reply pallets/protos/src/lib.rs:848]: So maybe we could leave that .filter by Categories::Trait and use your new structure `<Traits<T>>::contains_key(&trait_id)`

Comment (sinkingsugar) [reply pallets/protos/src/lib.rs:848]: but name is not available anymore, you can pass ID still no?

Comment (sinkingsugar) [reply pallets/protos/src/lib.rs:848]: basically now we would pass right IDs

Comment (alebaffa) [reply pallets/protos/src/lib.rs:848]: Oh yes I see, just by matching the objects will work now since there is only one field. Ok!

**Review by alebaffa — APPROVED** [review]

**Comment by sinkingsugar** [comment]
Thank you @alebaffa 🙏👍👍👍👍

**Pull request merged**